### DISCUSSION TITLE
How can I query all item under a category or in his childs (subcategory) ?

### DISCUSSION THREAD

**Discussion by NicoSan20** [sequelize/sequelize#14589] [Help & Questions] [answered]
Hello everyone !

Currently I have a hierarchical table "categories" like this :
```
id    name           parentid
----------------------------
1     animal         NULL
2     doggie         1
3     kittie         1
4     companion      2
5     herding        2
6     chihuahua      4
7     poodle         4
8     collie         5
9     shepherd       5
```

That's what I use to generate a categories section like below :
```
- animal (category id 1)
     - doggie (category id 2)
          - companion (category id 4)
               - chihuahua (category id 6)
               - poodle (category id 7)
          - herding (category id 5)
               - collie (category id 8)
               - shepherd (category id 9)
     - kittie (category id 3)
```
And in my items table, I have the category id corresponding to it, example :
```
id    name          categoyid    ...
------------------------------------
1     snoppy        6            ...
```

Now the purpose is to find all item under a category or in his childs (subcategory). Example :

If I select the category id 2 (doggie), I want to find all items with category id 2 and also all in his childs category (4, 5, 6, 7, 8, 9).

How to do this simply ? What is the best way to proceed in this type of case ?

Any help would be appreciated, thank you in advance.

**Comment by ephys** [comment, marked as answer]
You can't do that using findAll in an optimal manner, you need a [recursive query](https://www.postgresql.org/docs/current/queries-with.html) for this. I recommend using `sequelize.query` for this, and writing your query manually

Edit: if not using postgres, it's also available in most dialects. See https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL
